sw/demo/main.c

```diff
@@ -33,16 +31,12 @@ int main(void) {
       puts("Hello World! ");
       puthex(last_elapsed_time);
       puts("   Input Value: ");
-      puthex(read_gpio(GPIO_IN_DBNC));
+      uint32_t in_val = read_gpio(GPIO_IN_DBNC);
+      puthex(in_val);
       putchar('\n');
 
-      // Cycling through green LEDs
-      set_output_bit(GPIO_OUT, cur_output_bit_index, cur_output_bit);
-      cur_output_bit_index++;
-      if (cur_output_bit_index >= 4) {
-        cur_output_bit_index = 0;
-        cur_output_bit = !cur_output_bit;
-      }
+      // Cycling through green LEDs when BTN0 is pressed
+      set_outputs(GPIO_OUT_SHIFT, in_val);
```

Comment:
Have you already thought about SW that works with the GPIO with *and* without the shift register functionality? In the lab, the participant will run the demo app first with the unmodified GPIO, then modify the GPIO to add the shift register, and then run the demo app again. I think it's reasonable to let the participant "flip a switch" in SW but I'd like to avoid a situation where they would have to change multiple lines of SW

Otherwise LGTM (only last commit in this PR reviewed, other commits reviewed in https://github.com/lowRISC/ibex-demo-system/pull/10).

Reply:
Thanks Andreas. That makes sense. Now I changed corresponding part in the PR and force-pushed it. Now the same demo example is provided with or without shift register functionality. The participants are only needed to "flip a switch" in SW as you suggested.

Reply:
Nice, thanks!